cpp/arcticdb/entity/type_utils.hpp

```diff
@@ -15,6 +15,11 @@ inline bool trivially_compatible_types(entity::TypeDescriptor left, entity::Type
     if(left == right)
         return true;
 
+    // Empty type is coercible to any type, and any type is coercible to the empty type.
+    if(is_empty_type(left.data_type()) || is_empty_type(right.data_type())) {
+        return true;
+    }
```

Comment:
This change should be removed, `trivially_compatible_types` is used to `memcpy` buffers around, which doesn't make sense with `None`-type columns.
e.g. https://github.com/man-group/ArcticDB/blob/master/cpp/arcticdb/version/version_core.cpp#L747